src/hepattn/models/maskformer.py

```diff
@@ -161,6 +164,16 @@ def forward(self, inputs: dict[str, Tensor]) -> dict[str, Tensor]:
             else:
                 attn_mask = None
 
+            if (
+                self.log_attn_mask
+                and (attn_mask is not None)
+                and (self.step_ % 1000 == 0)
+            ):
+                # Store for callback to log later
+                self._last_attn_mask = attn_mask[0].detach().cpu().clone()
+                self._last_attn_mask_step = self.step_
+                self._last_attn_mask_layer = layer_index
```

Comment:
it looks like we will always log the final layer mask at the moment? is it worth logging e.g. the first and last?